### PR TITLE
[SofaNG] Setup Mapping 

### PR DESCRIPTION
**Pull request by fredroy**
2nd big guy

#1527
based on 
- ~~#2612 ~~
- #2717 (for a moved file in core)

Deprecates **SofaMiscMapping**
Partially deprecates **SofaBaseMechanics**, **SofaRigid**, **SofaGeneralRigid**

This module gathers all "mechanical" mappings (inhering BaseMapping) and __not__ topological mapping.
~~I am a bit concerned by the size of this module but I cannot really find any good sub-modules to sort these components into ?~~

Following a discussion with @ChristianDuriez @Younesssss @alxbilger , Mappings are organized between:
 - NonLinear: mappings implementing getK() and applyDJT()
 - Linear: the rest of mappings
 - MappedMatrix: components which are not mapping? but helps to map matrices (stiffness and stuff) 
      -> Ad-hoc module which is only used to patch up matrices in SOFA, will disappear once matrix assembly will be pushed

~~TODO (in an other PR), set Articulated-related things into a dedicated plugin.~~ Done in #2684 

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
🧐 `macos_clang_options — Build OK. FIXME: 26 units, 10 scenes, 1 regression `
.....
`ModuleNotFoundError: No module named numpy` 🤫

**Comment by guparan** [comment]
Problem fixed, build restarted :ok_hand:

File: Component/Mapping/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,145 @@
+cmake_minimum_required(VERSION 3.12)
+project(Sofa.Component.Mapping LANGUAGES CXX)
+
+set(SOFACOMPONENTMAPPING_SOURCE_DIR "src/sofa/component/mapping")
+
+set(HEADER_FILES
+    ${SOFACOMPONENTMAPPING_SOURCE_DIR}/config.h.in
+    ${SOFACOMPONENTMAPPING_SOURCE_DIR}/init.h
+    ${SOFACOMPONENTMAPPING_SOURCE_DIR}/BarycentricMapping.h
```
Comment (alxbilger) [inline Component/Mapping/CMakeLists.txt]: I am used to write the files in alphabetic order in a CMakeLists.txt.

**Comment by adagolodjo** [comment]
> On this pull-request we would need the feedback of @Younesssss and @ChristianDuriez about how to sort the mapping components in SOFA? How should be differentiated the different kind of mappings?
> 
> Your input is important to us for NG.

I can organize a meeting with @ChristianDuriez, @guparan, @fredroy around this subject. Maybe it's better that way?

File: Component/Mapping/CMakeLists.txt (hunk 1 above)
Comment (fredroy) [reply Component/Mapping/CMakeLists.txt]: For these 2 files in particular, I prefer to put them at the beginning, as there are not related to components (just to setup)

**Comment by fredroy** [comment]
> > On this pull-request we would need the feedback of @Younesssss and @ChristianDuriez about how to sort the mapping components in SOFA? How should be differentiated the different kind of mappings?
> > Your input is important to us for NG.
> 
> I can organize a meeting with @ChristianDuriez, @guparan, @fredroy around this subject. Maybe it's better that way?

It would be great yes !

**Pull request merged**